### PR TITLE
Make count NOT create a cursor

### PR DESCRIPTION
**Pull request by znewsham**
currently, calls to `Collection.find().count()` create a new session that is never released. This is because the mongo API changed when a session is created to init time of the cursor, and the cursor methods (fetch, forEach, map, count) all create a real cursor at call time. 

In the case of count, this cursor is never used.

I believe this is the fix to: https://github.com/meteor/meteor/issues/12321

### CODE REVIEW

File: packages/mongo/mongo_driver.js [hunk 1]
```diff
@@ -907,12 +907,27 @@ function setupSynchronousCursor(cursor, method) {
...
+  if (methodName === 'count') {
+    return;
+  }
   Cursor.prototype[methodName] = function (...args) {
     const cursor = setupSynchronousCursor(this, methodName);
     return cursor[methodName](...args);
   };
 
+
+  Cursor.prototype.count = function () {
```
Comment (baileympearson) [inline packages/mongo/mongo_driver.js:923]: This should go outside of the `forEach` loop, or inside the `methodName === 'count'` guard, so `Cursor.prototype.count` is not redefined on every iteration of the loop.

Comment (znewsham) [reply packages/mongo/mongo_driver.js:923]: :facepalm: thanks for the catch!

**Comment by Twisterking** [comment]
Thanks a lot for the very quick reaction! Any idea how quickly, usually, patches like this are pushed to a new (patch) version of Meteor?

**Comment by Grubba27** [comment]
I will create a patch for this ASAP will be released under the 2.8.2 namespace

**Comment by Grubba27** [comment]
Travis CI tests are failing

**Comment by znewsham** [comment]
@Grubba27 can you provide any details? that output is hard to parse - I ran the mongo tests locally and they all passed

**Comment by znewsham** [comment]
ah - is it this?

```
S: tinytest - livedata connection - reconnect to a different server failed: {"groupPath":["tinytest","livedata connection"],"test":"reconnect to a different server","events":[{"sequence":2477,"type":"fail","details":{"type":"assert_equal","expected":"[true]","actual":"[false]","not":false,"asyncBlock":0}}],"server":true}
```

**Comment by Grubba27** [comment]
yeah is that one. I will try running locally like this:

```bash
./meteor self-test
```

**Comment by Grubba27** [comment]
ah got it. Will run them locally if they all pass I will continue with release

**Comment by Grubba27** [comment]
All mongo tests have passed and the test that has not passes is a problem in our end reverse.meteor.com is down

**Comment by Grubba27** [comment]
2.8.2 was launched
check if all is okay there

**Comment by Grubba27** [comment]
Good to hear @DavidSichau, we released yesterday 2.9. Maybe will help even more te mem consumption
